.github/workflows/tests.yml

```
@@ -17,10 +17,53 @@ env:
   MAVEN_OPTS: "-Djansi.force=true -Dhttps.protocols=TLSv1.2 -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=WARN -Djava.awt.headless=true -XX:ThreadStackSize=1m"
 
 jobs:
+  check-should-run:
+    if: github.event_name == 'merge_group'
+    runs-on: ubuntu-latest
+    outputs:
+      SHOULD_RUN_FOR_MERGE_QUEUE: ${{ steps.check.outputs.SHOULD_RUN_FOR_MERGE_QUEUE }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Fetch integration branch
+        run: git fetch origin integration:integration
+
+      - name: Check if the merge_group actions should run
+        id: check
+        env:
+          GH_TOKEN: ${{ github.token }}
+        run: |
+          # Get the merge base between current branch and integration
+          MERGE_BASE=$(git merge-base HEAD integration)
+          # Get the current commit of integration
+          integration_HEAD=$(git rev-parse integration)
+          
+          # Get the number of PRs in the merge queue
+          QUEUE_SIZE=$(gh api graphql -f query='
+            query($owner:String!, $name:String!) {
+              repository(owner:$owner, name:$name) {
+                mergeQueue {
+                  entries {
+                    totalCount
+                  }
+                }
+              }
+            }' -f owner=${{ github.repository_owner }} -f name=${{ github.event.repository.name }} --jq '.data.repository.mergeQueue.entries.totalCount')
+          
+          if [ "$MERGE_BASE" != "$integration_HEAD" ] && [ "$QUEUE_SIZE" -le 1 ]; then
+            echo "SHOULD_RUN_FOR_MERGE_QUEUE=true" >> $GITHUB_OUTPUT
+          else
+            echo "SHOULD_RUN_FOR_MERGE_QUEUE=false" >> $GITHUB_OUTPUT
+          fi
   # Runs the pom sorter and code formatter to ensure that the code
   # is formatted and poms are sorted according to project rules. If changes are found,
   # they are committed back to the branch
   check-code-formatting:
+    needs: check-should-run
+    if: github.event_name != 'merge_group' || needs.check-should-run.outputs.SHOULD_RUN_FOR_MERGE_QUEUE == 'false'
```

```suggestion
    if: github.event_name == 'merge_group' && needs.check-should-run.outputs.SHOULD_RUN_FOR_MERGE_QUEUE == 'true'
```
At the moment this logic seems to be doing the opposite of what we would want. It is skipping the tests when they shouldn't be. With the suggested change, I think it would run on pull requests and pushes to PRs & branches, just not when the condition `[ "$MERGE_BASE" != "$integration_HEAD" ] && [ "$QUEUE_SIZE" -le 1 ];` is met and the event is `merge_group`. If this assessment is correct, this will have to change for all of the jobs using the `needs: check-should-run`

---

Good catch. Originally i had done BRANCH_IS_UP_TO_DATE, but I changed it when I added checking for merge queue length. Applying your suggestion